File: php/src/Account.php
<?php

/**
 * Account endpoints.
 *
 * Copyright (C) 2025 Cloze SDK Contributors
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

namespace Cloze\SDK;

class Account
{
    private $client;

    public function __construct(ClozeClient $client)
    {
        $this->client = $client;
    }

    /**
     * Get custom fields.
     *
     * @param string|null $relationtype Optional filter - 'person', 'project', or 'company' (or null for all)
     * @return array List of custom fields
     * @throws \Cloze\SDK\Exceptions\ClozeAPIError
     * @throws \Cloze\SDK\Exceptions\ClozeAuthenticationError
     * @throws \Cloze\SDK\Exceptions\ClozeRateLimitError
     */
    public function getFields(?string $relationtype = null): array
    {
        $params = [];
        if ($relationtype) {
            $params['relationtype'] = $relationtype;
        }

        return $this->client->makeRequest('GET', '/v1/user/fields', $params);
    }

    /**
     * Get user profile.
     *
     * @return array User profile information
     * @throws \Cloze\SDK\Exceptions\ClozeAPIError
     * @throws \Cloze\SDK\Exceptions\ClozeAuthenticationError
     * @throws \Cloze\SDK\Exceptions\ClozeRateLimitError
     */
    public function getProfile(): array
    {
        return $this->client->makeRequest('GET', '/v1/user/profile');
    }

    /**
     * Get people contact segments.
     *
     * @return array List of people segments
     * @throws \Cloze\SDK\Exceptions\ClozeAPIError
     * @throws \Cloze\SDK\Exceptions\ClozeAuthenticationError
     * @throws \Cloze\SDK\Exceptions\ClozeRateLimitError
     */
    public function getSegmentsPeople(): array
    {
        return $this->client->makeRequest('GET', '/v1/user/segments/people');
    }

    /**
     * Get project segments.
     *
     * @return array List of project segments
     * @throws \Cloze\SDK\Exceptions\ClozeAPIError
     * @throws \Cloze\SDK\Exceptions\ClozeAuthenticationError
     * @throws \Cloze\SDK\Exceptions\ClozeRateLimitError
     */
    public function getSegmentsProjects(): array
    {
        return $this->client->makeRequest('GET', '/v1/user/segments/projects');
    }

    /**
     * Get people contact stages.
     *
     * @return array List of people stages
     * @throws \Cloze\SDK\Exceptions\ClozeAPIError
     * @throws \Cloze\SDK\Exceptions\ClozeAuthenticationError
     * @throws \Cloze\SDK\Exceptions\ClozeRateLimitError
     */
    public function getStagesPeople(): array
    {
        return $this->client->makeRequest('GET', '/v1/user/stages/people');
    }

    /**
     * Get project stages.
     *
     * @return array List of project stages
     * @throws \Cloze\SDK\Exceptions\ClozeAPIError
     * @throws \Cloze\SDK\Exceptions\ClozeAuthenticationError
     * @throws \Cloze\SDK\Exceptions\ClozeRateLimitError
     */
    public function getStagesProjects(): array
    {
        return $this->client->makeRequest('GET', '/v1/user/stages/projects');
    }

    /**
     * Get steps.
     *
     * @return array List of steps
     * @throws \Cloze\SDK\Exceptions\ClozeAPIError
     * @throws \Cloze\SDK\Exceptions\ClozeAuthenticationError
     * @throws \Cloze\SDK\Exceptions\ClozeRateLimitError
     */
    public function getSteps(): array
    {
        return $this->client->makeRequest('GET', '/v1/user/steps');
    }

    /**
     * Get views and audiences.
     *
     * @return array List of views and audiences
     * @throws \Cloze\SDK\Exceptions\ClozeAPIError
     * @throws \Cloze\SDK\Exceptions\ClozeAuthenticationError
     * @throws \Cloze\SDK\Exceptions\ClozeRateLimitError
     */
    public function getViews(): array
    {
        return $this->client->makeRequest('GET', '/v1/user/views');
    }
}

File: python/cloze_sdk/account.py
"""Account endpoints for the Cloze API."""

from typing import Dict, Any, Optional


class Account:
    """Account endpoints."""
    
    def __init__(self, client):
        self.client = client
    
    def get_fields(self, relationtype: Optional[str] = None) -> Dict[str, Any]:
        """
        Get custom fields.
        
        Args:
            relationtype: Optional filter - 'person', 'project', or 'company' (or empty for all)
            
        Returns:
            List of custom fields
        """
        params = {}
        if relationtype:
            params["relationtype"] = relationtype
        
        return self.client._make_request(
            "GET",
            "/v1/user/fields",
            params=params
        )
    
    def get_profile(self) -> Dict[str, Any]:
        """
        Get user profile.
        
        Returns:
            User profile information
        """
        return self.client._make_request(
            "GET",
            "/v1/user/profile"
        )
    
    def get_segments_people(self) -> Dict[str, Any]:
        """
        Get people contact segments.
        
        Returns:
            List of people segments
        """
        return self.client._make_request(
            "GET",
            "/v1/user/segments/people"
        )
    
    def get_segments_projects(self) -> Dict[str, Any]:
        """
        Get project segments.
        
        Returns:
            List of project segments
        """
        return self.client._make_request(
            "GET",
            "/v1/user/segments/projects"
        )
    
    def get_stages_people(self) -> Dict[str, Any]:
        """
        Get people contact stages.
        
        Returns:
            List of people stages
        """
        return self.client._make_request(
            "GET",
            "/v1/user/stages/people"
        )
    
    def get_stages_projects(self) -> Dict[str, Any]:
        """
        Get project stages.
        
        Returns:
            List of project stages
        """
        return self.client._make_request(
            "GET",
            "/v1/user/stages/projects"
        )
    
    def get_steps(self) -> Dict[str, Any]:
        """
        Get steps.
        
        Returns:
            List of steps
        """
        return self.client._make_request(
            "GET",
            "/v1/user/steps"
        )
    
    def get_views(self) -> Dict[str, Any]:
        """
        Get views and audiences.
        
        Returns:
            List of views and audiences
        """
        return self.client._make_request(
            "GET",
            "/v1/user/views"
        )

